src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -168,21 +171,103 @@ trait EntityComponent {
       allEntitiesAction.flatMap(cloneEntities(destWorkspaceContext, _))
     }
 
+    def insertEntitiesSql(destWorkspaceContext: SlickWorkspaceContext, entities: Map[Entity, UUID]) = {
+      val workspaceId = destWorkspaceContext.workspaceId
+
+      val insertBatches = entities.map { case (entity, entityId) =>
+        sql"($entityId, ${entity.name}, ${entity.entityType}, $workspaceId)"
+      }.grouped(1000).toSeq
+
+      DBIO.seq(insertBatches.map { insertBatch =>
+        val prefix = sql"insert into ENTITY (id, name, entity_type, workspace_id) values "
+        val suffix = insertBatch.reduce { (a, b) =>
+          concatSqlActionsWithDelim(a, b, ",")
+        }
+        concatSqlActions(prefix, suffix).as[Int]
+      }:_*)
+    }
+
+    def insertAttributesSql(destWorkspaceContext: SlickWorkspaceContext, attrsWithEntityIdMap: Seq[(UUID, (String, Attribute), UUID, Option[Int])]) = {
```

Comment:
Does the sql interpolator not know how to handle Options? I think it should line up and if it can't handle Options it should be pretty easy to write the parameter setter that can.

On the marshalAttribute front, all the code to do the right thing is in insertAttributeRecords except that code also does the insert. Extract out the marshal functionality. It would be good if we had 1 and only 1 way to insert attributes. I don't think this code handles lists appropriately.